llvm/tools/sycl-post-link/SYCLDeviceRequirements.cpp

```diff
@@ -21,11 +21,18 @@ using namespace llvm;
 void llvm::getSYCLDeviceRequirements(
     const module_split::ModuleDesc &MD,
     std::map<StringRef, util::PropertyValue> &Requirements) {
-  auto ExtractIntegerFromMDNodeOperand = [=](const MDNode *N,
-                                             unsigned OpNo) -> int32_t {
+  auto ExtractSignedIntegerFromMDNodeOperand = [=](const MDNode *N,
+                                                   unsigned OpNo) -> auto {
```

Comment:
I would prefer to list a concrete return type instead of `-> auto` here, because it affects readability of the code. [Coding standards](https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable):

> Use auto if and only if it makes the code more readable or easier to maintain. Don’t “almost always” use auto

In this particular case, I don't think that the return type is obvious from a context.

Reply:
Indeed, fixed.